src/ranged.cpp

```diff
@@ -1218,6 +1218,23 @@ void practice_archery_proficiency( player &p, const item &relevant )
     }
 }
 
+// Apply stamina cost to archery which decreases due to proficiency
+static void mod_stamina_archery( player &p, const item &relevant )
+{
+    // Set activity level to 10 * str_ratio, with 10 being max (EXTRA_EXERCISE)
+    // This ratio should never be below 0 and above 1
+    const int str_ratio =  relevant.get_min_str() / p.str_cur;
+    p.set_activity_level( 10 * str_ratio );
```

Comment:
Integer math doesn't work here due to truncation.
str_ratio will be 1 when str_cur == get_min_str and 0 if str_cur > get_min_str.
```suggestion
    // Set activity level to 10 * str_ratio, with 10 being max (EXTRA_EXERCISE)
    // This ratio should never be below 0 and above 1
    const int scaled_str_ratio =  ( 10 * relevant.get_min_str() ) / p.str_cur;
    p.set_activity_level( scaled_str_ratio );
```

Reply:
Oops, I actually meant for that to be a double not an int, but that works too. Thanks for the review!